packages/auth/src/providers/cognito/apis/setUpTOTP.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { AuthError } from '../../../errors/AuthError';
+import { TOTPSetupDetails } from '../../../types/models';
+import { SETUP_TOTP_EXCEPTION,  AssociateSoftwareTokenException } from '../types/errors';
+import { associateSoftwareTokenClient } from '../utils/clients/AssociateSoftwareTokenClient';
+import { getTOTPSetupDetails } from '../utils/signInHelpers';
+/**
+ * Sets up TOTP for the user.
+ * @throws  service error thrown when setting up TOTP: {@link AssociateSoftwareTokenException}.
```

Comment:
Nit: Technically I think the format should look like this according to TSDocs, but if it looks good in your IDE should be fine.

```suggestion
 *
 * @throws {@link AssociateSoftwareTokenException}
 * Thrown is a service occurs while setting up TOTP.
```